pkg/signatures/public_key_fetcher.go

```diff
@@ -124,15 +125,17 @@ func (c *cosignPublicKeySignatureFetcher) FetchSignatures(ctx context.Context, i
 // Note: This takes into account the definition of the transport.Error, you can find more here:
 // https://github.com/google/go-containerregistry/blob/f1fa40b162a1601a863364e8a2f63bbb9e4ff36e/pkg/v1/remote/transport/error.go#L90
 func makeTransientErrorRetryable(err error) error {
+	var transportErr *transport.Error
+	var urlError *url.Error
 	// We don't expect any transient errors that are coming from cosign at the moment.
-	if transportErr, ok := err.(*transport.Error); ok && transportErr.Temporary() {
-		return retry.MakeRetryable(err)
+	if errors.As(err, &transportErr) && transportErr.Temporary() {
+		return retry.MakeRetryable(transportErr)
```

Comment:
Why do you return the unwrapped error here, instead of `err`?

Reply:
Ah, good catch! You're right, we shouldn't return the unwrapped error here.